lib/Zonemaster/Backend/TestAgent.pm

```diff
@@ -158,6 +102,56 @@ sub run {
         Zonemaster::Engine::Profile->effective->set( q{net.ipv6}, ( $params->{ipv6} ) ? ( 1 ) : ( 0 ) );
     }
 
+    my %methods = Zonemaster::Engine->all_methods;
+
+    # BASIC methods are always run
+    $counter_for_progress_indicator{planned} = {
+        'Basic::basic00' => 1,
+        'Basic::basic01' => 1,
+        'Basic::basic02' => 1,
+        'Basic::basic03' => 1,
+        'Basic::basic04' => 1,
+    };
+    foreach my $module ( keys %methods ) {
+        foreach my $method ( @{ $methods{$module} } ) {
+            if ( Zonemaster::Engine::Util::should_run_test( $method ) ) {
+                $counter_for_progress_indicator{planned}{ $module . '::' . $method } = 1;
+            }
+        }
+    }
+
+    my @planned_methods = sort keys %{ $counter_for_progress_indicator{planned} };
+
+    # used for progress indicator
+    my ( $previous_module, $previous_method ) = ( '', '' );
+
+    if ( $show_progress ) {
+        # Callback defined here so it closes over the setup above.
+        Zonemaster::Engine->logger->callback(
+            sub {
+                my ( $entry ) = @_;
+
+                if ( $entry->{tag} and $entry->{tag} eq 'TEST_CASE_END' ) {
+                    my $current_module = ucfirst( lc $entry->{_module} );
+                    my $current_testcase = $entry->{args}->{testcase};
+                    my $current_module_method = $current_module . "::" . $current_testcase;
+
+                    $counter_for_progress_indicator{executed}{$current_module_method}++;
```

Comment:
Is there an advantage to use a hash to keep track of the number of planned / executed tests instead of simply a int?

Reply:
Very good remark, I kept this part from the old code without too much thinking. There is no point in using a hash anymore. I'll provide an update.